### PR TITLE
k8s: Treat service annotation changes as unequal objects

### PR DESCRIPTION
**Pull request by tgraf**
Fixes: #6393

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6407)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/k8s/factory_functions.go [hunk 1]
```diff
@@ -419,6 +419,10 @@ func equalV1Services(o1, o2 interface{}) bool {
 		return false
 	}
 
+	if !reflect.DeepEqual(svc1.GetAnnotations(), svc2.GetAnnotations()) {
```
Comment (aanm) [inline pkg/k8s/factory_functions.go]: Please write a comment specifying why we use the annotations. also you have this available
```
comparator.MapStringEquals(svc1.GetAnnotations(), svc2.GetAnnotations())
```

**Review by aanm — DISMISSED** [review]
If the annotations contain logic on how a service should behave, why aren't we storing the annotations in `pkg/k8s/service.go:Service`?

**Comment by tgraf** [comment]
> If the annotations contain logic on how a service should behave, why aren't we storing the annotations in `pkg/k8s/service.go:Service`?

We are accessing the Annotations via the stored k8s object.

```
func getAnnotationIncludeExternal(svc *v1.Service) bool {
        if value, ok := svc.ObjectMeta.Annotations[annotation.GlobalService]; ok {
                return strings.ToLower(value) == "true"
        }

        return false
}
```

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**